parsing/unit_info.mli

```diff
@@ -0,0 +1,136 @@
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** This module centralize the handling of compilation files and their metadata.
+
+  Maybe more importantly, this module provides functions for deriving module
+  names from strings or filenames.
+*)
+
+type ('source,'target) t
+(**  A value [file] of type {!t} records
+      - the module name associated to the file
+      - the target filename or prefix
+      - the source file for compilation file if it exists
+*)
+
+(** Source or partial target file *)
+
+type source_file = (string, unit) t
+type target_prefix = (string, unit) t
+(** The prefix of a target file with its module name and source file *)
+
+(** Target subfamily *)
+type 'a any_target = ('a, string) t
+
+type target = string any_target
+type target_only = unit any_target
+(** A target file without a source file: this typically happen when packing *)
+
+(** [source_file u] returns the source file of [u] *)
+val source_file: (string,_) t -> string
+
+(** [source_file_opt u] returns the source file of [u] if it exists *)
+val source_file_opt: _ t -> string option
+
+(** [prefix u] returns the prefix of the target. *)
+val prefix: _ t -> string
+
+(** [filename u] returns the file name of the target. *)
+val filename: _ any_target -> string
+
+(** [modname u] returns the module name of the compilation file.*)
+val modname: _ t -> string
+
+(** {1:modname_from_strings Module name computation} *)
+
+(** [modulize s] capitalizes the first letter of [s]. *)
+val modulize: string -> string
+
+(** [modname_from_source filename] is [modulize stem] where [stem] is the
+    basename of the filename [file] stripped from all its extensions.*)
+val modname_from_source: string -> string
+
+(** {2:module_name_validation Module name validation function}*)
+
+(** [is_unit_name ~strict name] returns true only if [name] can be used as a
+    correct module name. The [strict] mode enforces that the first character is
+    an ascii capital letter. *)
+val is_unit_name : strict:bool -> string -> bool
+
+(** [check_unit_name file] prints a warning if the derived module name [modname
+    file] should not be used as a module name. *)
+val check_unit_name : (string,_) t -> unit
+
+(** {1:info_creation Creation of file metadata} *)
+
+(** [from_source filename] associates the module name [modname_from_source
+    filename] to the source file [filename]. *)
+val from_source: string -> source_file
```

Comment:
These docstrings are a bit confusing because the type is `path -> source_file`, but the description is about module names. I wonder if a more abstract description would make sense, for example:

```ocaml
(** [from_source filename] is the unit information for the source file [filename]. *)
```